src/Workspaces/Core/Portable/Rename/ConflictEngine/DeclarationConflictHelpers.cs

```diff
@@ -58,51 +61,43 @@ public bool Equals(List<ITypeSymbol> x, List<ITypeSymbol> y)
                 return x.SequenceEqual(y);
```

Comment:
If this is a perf sensitive area then shouldn't we be avoiding this method?  This will at the very least box the `ImmutableArray<T>` instance to `IEnumerable<T>` and allocate enumerators.  

Or does this go to a specialized override for immutable arrays.